fighters/dedede/src/status/special_hi.rs

```diff
@@ -3,10 +3,15 @@ use super::*;
 // FIGHTER_DEDEDE_STATUS_KIND_SPECIAL_HI_FAILURE
 
 unsafe extern "C" fn special_hi_failure_main(fighter: &mut L2CFighterCommon) -> L2CValue{
```

Comment:
this main status should turn into a pre one that looks similar to this
```rs
unsafe extern "C" fn special_hi_failure_main(fighter: &mut L2CFighterCommon) -> L2CValue{
    if !VarModule::is_flag(fighter.battle_object, vars::dedede::instance::JET_GROUND_BONK) {
        StatusModule::set_status_kind_interrupt(fighter.module_accessor, *FIGHTER_STATUS_KIND_FALL_SPECIAL);
        return 1.into();
    }
    smashline::original_status(Pre, fighter, *FIGHTER_DEDEDE_STATUS_KIND_SPECIAL_HI_FAILURE)(fighter)
}
```

See below notes as to why.